xblockutils/public/studio_container.js

```diff
@@ -0,0 +1,58 @@
+function StudioContainerXBlockWithNestedXBlocksMixin(runtime, element) {
+    var $buttons = $(".add-xblock-component-button", element),
+        $element = $(element);
+
+    function isSingleInstance($button) {
+        return $button.data('single-instance');
+    }
+
+    $buttons.click(function(ev) {
+        var $this = $(this);
+        if ($this.is('.disabled')) {
+            ev.preventDefault();
+            ev.stopPropagation();
+        } else {
+            if (isSingleInstance($this)) {
+                $this.addClass('disabled');
+                $this.attr('disabled', 'disabled');
+            }
+        }
+    });
+
+    function updateButtons() {
+        var nestedBlockLocations = $.map($element.find(".studio-xblock-wrapper"), function(block_wrapper) {
+           return $(block_wrapper).data('locator');
+        });
+
+        $buttons.each(function() {
+            var $this = $(this);
+            if (!isSingleInstance($this)) {
+                return;
+            }
+            var category = $this.data('category');
+            var childExists = false;
+
+            // FIXME: This is potentially buggy - if some XBlock's category is a substring of some other XBlock category
```

Comment:
Yes, but it also means none of those people necessarily feel any obligation to actually fix it if they can avoid it. :) That's why we make tasks for things that are fixmes, to make sure they get fixed instead of threatening to surprise expand someone's scope.